src/main/kotlin/com/linchpino/core/repository/AccountRepository.kt

```diff
@@ -1,8 +1,50 @@
 package com.linchpino.core.repository
 
+import com.linchpino.core.dto.MentorWithClosestTimeSlot
 import com.linchpino.core.entity.Account
 import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
 import org.springframework.stereotype.Repository
+import java.time.LocalDate
 
 @Repository
-interface AccountRepository : JpaRepository<Account, Long>
\ No newline at end of file
+interface AccountRepository : JpaRepository<Account, Long>{
+
+    @Query(
+        """
+    SELECT NEW com.linchpino.core.dto.MentorWithClosestTimeSlot(
+            a.id ,
+            a.firstName,
+            a.lastName,
+            mts.id,
+            mts.fromTime,
+            mts.toTime
+        )
+    FROM
+        Account a
+    JOIN
+        a.interviewTypes it
+    JOIN
+        MentorTimeSlot mts ON mts.account.id = a.id
+    WHERE
+        a.type = 'MENTOR'
+        AND mts.date = :dateParam
+        AND it.id = :interviewTypeId
+        AND mts.fromTime = (
+            SELECT
+                MIN(mts2.fromTime)
+            FROM
+                MentorTimeSlot mts2
+            WHERE
+                mts2.account.id = a.id
+                AND mts2.date = :dateParam
+                AND mts2.status = 'AVAILABLE'
```

Comment:
Do you mean we should use ordinal instead of string to save enums? or add a constructor for the enums to take an integer value and write a custom converter for hibernate for mapping enum to it's corresponding value?

Reply:
Yes, I mean add a constructor for the enums to take an integer value and write a custom converter for hibernate for mapping enum to it's corresponding value. Totally in the code it's better to to work on enum elements and in the db use int value of it to keep space